dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeFactoryImpl.java

```diff
@@ -437,6 +437,11 @@ private ContentType dbSaveUpdate(final ContentType saveType) throws DotDataExcep
    * @param fieldVariables original field variables
    */
   private void saveFieldVariables(final Field field, List<FieldVariable> fieldVariables) {
+      if (field.id() == null) {
+          Logger.warn(getClass(), String.format("Found null id at field %s", field.name()));
```

Comment:
I'd add "Not saving field variables. Found null...."

Reply:
Done